openstack/standalone-cinder/pkg/provisioner/provisioner.go

```diff
@@ -74,19 +77,37 @@ func NewCinderProvisioner(client kubernetes.Interface, id, configFilePath string
 // Provision creates a storage asset and returns a PV object representing it.
 func (p *cinderProvisioner) Provision(options controller.VolumeOptions) (*v1.PersistentVolume, error) {
 	var (
-		connection volumeservice.VolumeConnection
-		mapper     volumeMapper
-		pv         *v1.PersistentVolume
-		cleanupErr error
+		connection        volumeservice.VolumeConnection
+		mapper            volumeMapper
+		pv                *v1.PersistentVolume
+		volumeID          string
+		err               error
+		cleanupErr        error
+		useExistingVolume bool
+		volumeState       string
 	)
 
 	if options.PVC.Spec.Selector != nil {
 		return nil, fmt.Errorf("claim Selector is not supported")
 	}
 
+	if volumeID, useExistingVolume = options.PVC.Annotations[ShouldProvisionVolumeAnn]; useExistingVolume {
```

Comment:
for better readability, add a var called sourceVolumeID and use it here. Use volumeID for new volume.